src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,115 @@
+import classNames from 'classnames';
+import React, {
+  useCallback,
+  useMemo,
+  useState,
+} from 'react';
+
+import { Person } from '../types/Person';
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+function debounce(callback: Function, delay: number) {
+  let timerId = 0;
+
+  return (...args: any) => {
+    window.clearTimeout(timerId);
+
+    timerId = window.setTimeout(() => {
+      callback(...args);
+    }, delay);
+  };
+}
+
+type Props = {
+  people: Person[];
+  onSelect?: React.Dispatch<React.SetStateAction<Person | null>>;
+  delay: number;
+  selectedPerson: Person | null;
+};
+
+export const Autocomplete: React.FC<Props> = React.memo(
+  ({
+    people,
+    onSelect = () => {},
+    delay,
+    selectedPerson,
+  }) => {
+    const [query, setQuery] = useState('');
+    const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+    const [appliedQuery, setAppliedQuery] = useState('');
+
+    const applyQuery = useCallback(() => debounce(
```

Comment:
useCallback is redundant here - anyway, you don't memorize other things that depends on this function

you should call `applyQuery(event.target.value)` in this way below